core/modules/book/book.theme.inc

```diff
@@ -46,23 +46,54 @@ function template_preprocess_book_navigation(&$variables) {
   $variables['tree'] = '';
 
   if ($book_link['mlid']) {
-    $variables['tree'] = book_children($book_link);
-
-    if ($prev = book_prev($book_link)) {
+    // use book_link to see if we can obtain info about it from cache
+    $data = FALSE;
+    // generate a unique call signature for bid and mlid
+    $cid = $book_link['bid'] . ':' . $book_link['mlid'];
+    // statically cache future calls
+    $data = &backdrop_static(__FUNCTION__ . $cid);
```

Comment:
Rather than making a different `backdrop_static()` for each cache entry, we should have a single static cache that is an array of values. e.g.
```
$data = NULL;
$cache = &backdrop_static(__FUNCTION__, array());
if (isset($cache[$cid])) {
  $data = $cache[$cid];
}
```